### PR TITLE
Search filter update

### PR DESCRIPTION
**Pull request by seromenho**

### CODE REVIEW

File: js/build-directory.js [hunk 1]
```diff
@@ -1032,18 +1032,29 @@ DataDirectory.prototype.renderSearchResult = function( options, callback ){
   if (typeof callback === 'function') setTimeout(callback, 0);
 };
 
-DataDirectory.prototype.search = function( value ) {
-  var path = ':root > :has(';
+DataDirectory.prototype.search = function(search) {
+  var entries = [];
   var searchFields = this.config.search_fields;
+  this.data.forEach(function (entry) {
+    for (var i = 0; i < searchFields.length; i++) {
```
Comment (tonytlwu) [inline js/build-directory.js:1039]: Use `.forEach()` too here

File: js/build-directory.js [hunk 2]
```diff
@@ -1032,18 +1032,29 @@ DataDirectory.prototype.renderSearchResult = function( options, callback ){
...
-    path += '."' + searchFields[i] + '":contains("' + value + '")';
-  }
-  path += ')';
+      if (typeof value === 'string' && value.match(term)) {
+        return entries.push(entry);
+      }
 
-  return JSONSelect.match( path, this.data );
-}
+      if (value === search) {
```
Comment (tonytlwu) [inline js/build-directory.js:1050]: Is this in case the search value is not a string, e.g. number or boolean?

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/build-directory.js (hunk 1 above)
Comment (seromenho) [reply js/build-directory.js:1039]: @tonytlwu I want to return as soon I found a match on any field. With forEach you can't short-circuit it. Unless you trow and catch the error (more painfull than using a for loop)

File: js/build-directory.js (hunk 2 above)
Comment (seromenho) [reply js/build-directory.js:1050]: Exactly.

**Review by tonytlwu — APPROVED** [review]

File: js/build-directory.js (hunk 1 above)
Comment (tonytlwu) [reply js/build-directory.js:1039]: Makes sense 👍

**Pull request merged**